core/codegen/tests/from_form.rs

```diff
@@ -618,3 +618,33 @@ fn test_nested_multi() {
 //     println!("String: {}", string);
 //     panic!(string);
 // }
+
+#[test]
+fn test_defaults() {
+    #[derive(FromForm, PartialEq, Debug)]
+    struct FormWithDefaults {
+        #[field(default = 100)]
+        field1: i32,
+        field2: i32,
+
+        #[field(default = true)]
+        field3: bool,
+        #[field(default = false)]
+        field4: bool,
+        field5: bool,
+    }
+
+    let form_string = &["field1=101", "field2=102"].join("&");
+
+    let form: Option<FormWithDefaults> = lenient(&form_string).ok();
+    assert_eq!(form, Some(FormWithDefaults {
+        field1: 101,
+        field2: 102,
+        field3: true,
+        field4: false,
+        field5: false,
+    }));
+    let form: Option<FormWithDefaults> = strict(&form_string).ok();
+    assert!(form.is_none());
+
```

Comment:
Add a test that specifies all of the fields and ensure that the strict parse works.